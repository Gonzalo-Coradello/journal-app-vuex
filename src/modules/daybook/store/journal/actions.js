import journalApi from '@/api/journalApi'

export const loadEntries = async ({ commit }) => {
  const { data } = await journalApi.get('/entries.json')

  if (!data) {
    commit('setEntries', [])
    return
  }

  const entries = []

  for (let id of Object.keys(data)) {
    entries.push({
      id,
      ...data[id]
    })
  }

  commit('setEntries', entries)
}

export const updateEntry = async ({ commit }, entry) => {
  const { id, ...entryData } = entry
  await journalApi.put(`/entries/${id}.json`, entryData)

  commit('updateEntry', { ...entry })
}

export const createEntry = async ({ commit }, entry) => {
  const { data } = await journalApi.post(`/entries.json`, entry)

  commit('addEntry', { id: data.name, ...entry })
  return data.name
}

export const deleteEntry = async ({ commit }, id) => {
  console.log({ id })

  await journalApi.delete(`/entries/${id}.json`)

  commit('deleteEntry', id)
}
