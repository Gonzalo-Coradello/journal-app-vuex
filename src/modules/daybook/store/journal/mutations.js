// export const myMutation = (state) => {
//
// }

export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries]
  state.isLoading = false
}

export const updateEntry = (state, entry) => {
  state.entries = state.entries.map((e) => (e.id === entry.id ? entry : e))
}

export const addEntry = (state, entry) => {
  state.entries.unshift(entry)
}

export const deleteEntry = (state, id) => {
  state.entries = state.entries.filter((entry) => entry.id !== id)
}
