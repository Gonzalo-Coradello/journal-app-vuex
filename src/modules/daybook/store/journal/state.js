export default () => ({
  isLoading: true,
  entries: [
    {
      id: '1',
      date: new Date().toDateString(),
      text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus rem at ducimus repellendus odio eos libero molestias officia illum temporibus commodi omnis, odit quas laudantium quo ipsa natus tempore magni.',
      picture: null,
    },
    {
      id: '2',
      date: new Date('2023/12/31').toDateString(),
      text: 'In, nobis similique! Voluptas, qui nobis maxime explicabo, laborum beatae consectetur laboriosam tempore dignissimos libero quas dolor voluptates odio temporibus? Voluptatum eligendi tempora nobis expedita qui provident perferendis quo reiciendis aperiam iure? Magni deleniti quaerat accusantium atque nisi!',
      picture: null,
    },
    {
      id: '3',
      date: new Date('2024/01/01').toDateString(),
      text: 'Repellendus consequatur atque accusamus iste vel sequi eveniet eos rem. Illum, ea? Rerum maxime quasi, quidem saepe temporibus facilis nesciunt sed nisi perspiciatis magni animi reprehenderit vitae error fuga, quaerat ad, excepturi assumenda quas autem enim!',
      picture: null,
    },
  ]
})
