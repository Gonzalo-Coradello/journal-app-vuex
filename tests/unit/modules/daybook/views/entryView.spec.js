import journal from '@/modules/daybook/store/journal'
import { shallowMount } from '@vue/test-utils'
import { describe, test, vi, beforeEach, expect } from 'vitest'
import { createStore } from 'vuex'
import { journalState } from '../../../mock-data/test-journal-state'
import EntryView from '@/modules/daybook/views/EntryView.vue'

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState }
      }
    }
  })

vi.mock('sweetalert2', async () => {
  const actual = await vi.importActual('sweetalert2')

  return {
    ...actual,
    fire: vi.fn(),
    showLoading: vi.fn(),
    close: vi.fn()
  }
})

describe('Pruebas en el EntryView', () => {
  const store = createVuexStore(journalState)
  const mockRouter = {
    push: vi.fn()
  }

  let wrapper

  beforeEach(() => {
    vi.clearAllMocks()

    wrapper = shallowMount(EntryView, {
      props: {
        id: '-NnJqfmqd1bxoKeE81Z-'
      },
      global: {
        mocks: {
          $router: mockRouter
        },
        plugins: [store]
      }
    })
  })

  test('debe sacar al usuario cuando el id no existe', () => {
    shallowMount(EntryView, {
      props: {
        id: 'Este id no existe'
      },
      global: {
        mocks: {
          $router: mockRouter
        },
        plugins: [store]
      }
    })

    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
  })

  test('debe mostrar la entrada correctamente', () => {
    expect(wrapper.html()).toMatchSnapshot()
    expect(mockRouter.push).not.toHaveBeenCalled()
  })
})
