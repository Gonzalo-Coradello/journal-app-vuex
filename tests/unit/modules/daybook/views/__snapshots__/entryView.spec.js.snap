// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en el EntryView > debe mostrar la entrada correctamente 1`] = `
"<div data-v-a5d136c6="" class="entry-title d-flex justify-content-between p-2">
  <div data-v-a5d136c6=""><span data-v-a5d136c6="" class="text-success fs-3 fw-bold">4</span><span data-v-a5d136c6="" class="mx-2 fs-3">Enero</span><span data-v-a5d136c6="" class="mx-2 fs-4 fw-light">2024, Jueves</span></div>
  <div data-v-a5d136c6=""><input data-v-a5d136c6="" type="file" accept="image/png, image/jpeg" style="display: none;"><button data-v-a5d136c6="" class="btn btn-danger mx-2"> Borrar <i data-v-a5d136c6="" class="fa fa-trash-alt"></i></button><button data-v-a5d136c6="" class="btn btn-danger mx-2"> Subir foto <i data-v-a5d136c6="" class="fa fa-upload"></i></button></div>
</div>
<hr data-v-a5d136c6="">
<div data-v-a5d136c6="" class="d-flex flex-column px-3 h-75"><textarea data-v-a5d136c6="" placeholder="¿Qué sucedió hoy?"></textarea></div>
<img data-v-a5d136c6="" src="https://res.cloudinary.com/dqkb4ydrr/image/upload/v1704403189/mztes7bijs0xhp8rd9jv.jpg" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
<fab-stub data-v-a5d136c6="" icon="fa-save"></fab-stub>"
`;
